src/main/java/cn/nukkit/Player.java

```diff
@@ -2513,7 +2512,7 @@ public void onCompletion(Server server) {
                     break;
                 case ProtocolInfo.PLAYER_SKIN_PACKET:
                     PlayerSkinPacket skinPacket = (PlayerSkinPacket) packet;
-                    Skin skin = skinPacket.skin;
+                    skin = skinPacket.skin;

```

Comment:
This sets the skin even if the `isValid` below doesn't pass?

Reply:
No, it should reference to [this instance](https://github.com/PetteriM1/NukkitPetteriM1Edition/blob/9a6996c2e3d44f306d57875f3abae414fbbcba76/src/main/java/cn/nukkit/Player.java#L2419).